### ISSUE TITLE
Search Buffer Symbols modal doesn't dismiss on tab close

### ISSUE THREAD

**Issue by Angelk90** [zed-industries/zed#21962]
### Summary

If I open "Search Buffer Symbols" and then close the buffer tab the modal is still open.

### Environment

```
Zed: v0.166.0 (Zed Preview)
OS: macOS 14.1.1
Memory: 16 GiB
Architecture: x86_64
```

**Comment by notpeter** [comment]
Yes.  The Outline Panel for empty files (or those without outline.scm tree-sitter results) includes the filename at the top as a placeholder so the Outline panel is not empty.
<img width="229" alt="Screenshot 2024-12-16 at 0 36 27" src="https://github.com/user-attachments/assets/fc59f740-ede4-47d1-ba1d-c8f6758d6bec" />

"Outline: Toggle" shows you the symbols in the current buffer, of which there aren't any.  This is as designed.

---

The "Outline Toggle" modal sticking around is a bug.
Steps to reproduce:
 
1. Open a file.
2. Toggle the outline (cmd-shift-o)
3. Click on the close icon in the file tab.

The Outline modal should have been dismissed, but it sticks around.
<img width="1235" alt="Screenshot 2024-12-16 at 0 38 02" src="https://github.com/user-attachments/assets/d65eab7a-8dd3-4f61-b7af-f72c5f4ff646" />

**Comment by mikalsande** [comment]
It looks like this has been fixed, tabs no longer show the x to close the tab when the "Search buffer symbols" modal is open. When the tab is clicked on while the modal is open the modal closes and the x to close the tab reappears. Also cannot close the tab with keyboard shortcut. And opening the Execute command modal cancels the Search buffer modal.

Environment
```
Zed: v0.201.8
OS: macOS 15.6.1
Memory: 36 GiB
Architecture: aarch64
```

<details>
  <summary>Screen capture</summary>
  
https://github.com/user-attachments/assets/aa1b3519-8771-4802-8fb3-6046b76deea6

</details>

**Comment by notpeter** [comment]
Thanks @mikalsande
